pwn: add page (#12352)
tldr-pages/tldr: pages/linux/pwn.md

# pwn

> Exploit Development Library designed for rapid prototyping.
> More information: <https://docs.pwntools.com/en/stable/commandline.html>.

- Convert given assembly code to `bytes`:

> `pwn asm {{'xor edi, edi'}}`

Reviewer: ```suggestion
`pwn asm "{{xor edi, edi}}"`
```

> - Create a cyclic pattern of `100` characters:

Reviewer: ```suggestion
- Create a cyclic pattern of 100 characters:
```

> `pwn cyclic {{100}}`

Reviewer: Should stay as-is in my opinion. The "100" can be changed by the user without really changing the meaning of the command, so it is a placeholder. We could change the "100" to "value" and also edit the description.

> - Create a cyclic pattern of 100 characters:
> 
> `pwn cyclic {{100}}`

Reviewer 2: ```suggestion
- Create a cyclic pattern of the specific number of characters:

`pwn cyclic {{value}}`
```

Author: I guess having {{value}} sounds more generic IMO since only numbers could be mentioned.

> - Create a cyclic pattern of 100 characters:
> 
> `pwn cyclic {{100}}`

Reviewer 3: ```suggestion
- Create a cyclic pattern of the specific number of characters:

`pwn cyclic {{number}}`
```

How about this?

> - Print `bytes` for the given `hex-encoded` data:

Reviewer: ```suggestion
- Encode the given data into the hexadecimal system:
```

`pwn hex {{deafbeef}}`

> - Print `hex-encoded` data for the given `bytes`:

Reviewer: ```suggestion
- Decode the given data from hexadecimal:
```

`pwn unhex {{6c4f7645}}`

> - Print `x64 Linux Shellcode` in bytes that runs a `shell`:

Reviewer: ```suggestion
- Print a x64 Linux shellcode for running a shell:
```

`pwn shellcraft {{amd64.linux.sh}}`

> - Check binary security settings for the given `ELF file`:

Reviewer: ```suggestion
- Check the binary security settings for the given ELF file:
```

> `pwn checksec {{program}}`

Reviewer: ```suggestion
`pwn checksec {{path/to/file}}`
```

> - Check for pwntools updates:

Reviewer: ```suggestion
- Check for Pwntools updates:
```

`pwn update`

> - Print Pwntools help menu:

Reviewer: ```suggestion
- Display version:
```
